src/python_testing/TC_SC_4_3.py

```diff
@@ -0,0 +1,378 @@
+            active_mode_threshold_ms = await self.get_idle_mode_threshhold_ms()
+        logging.info(f"\n\n\tactive_mode_threshold_ms: {active_mode_threshold_ms}\n\n")
+
+        # *** STEP 4 ***
+        # If supports_icd is true, TH reads FeatureMap from the ICD Management cluster on EP0. If the LITS feature
+        # is set, set supports_lit to true. Otherwise set supports_lit to false.
+        self.step(4)
+        if supports_icd:
+            feature_map = await self.get_icd_feature_map()
+            LITS = Clusters.IcdManagement.Bitmaps.Feature.kLongIdleTimeSupport
+            supports_lit = bool(feature_map & LITS == LITS)
+            logging.info(f"\n\n\tkLongIdleTimeSupport set: {supports_lit}\n\n")
+
+        # *** STEP 5 ***
+        # TH constructs the instance name for the DUT as the 64-bit compressed Fabric identifier, and the
+        # assigned 64-bit Node identifier, each expressed as a fixed-length sixteen-character hexadecimal
+        # string, encoded as ASCII (UTF-8) text using capital letters, separated by a hyphen.
+        self.step(5)
+        instance_name = self.get_dut_instance_name()
+        instance_qname = f"{instance_name}.{MdnsServiceType.OPERATIONAL.value}"
+
+        # *** STEP 6 ***
+        # TH performs a query for the SRV record against the qname instance_qname.
+        # Verify SRV record is returned
+        self.step(6)
+        print("\n"*10)
+        mdns = MdnsDiscovery()
+        operational_record = await mdns.get_service_by_record_type(
+            service_name=instance_qname,
+            service_type=MdnsServiceType.OPERATIONAL.value,
+            record_type=DNSRecordType.SRV,
+            log_output=True,
+            load_from_cache=False
+        )
+        
+        # Verify SRV record is returned
+        srv_record_returned = operational_record is not None and operational_record.service_name == instance_qname
+        asserts.assert_true(srv_record_returned, "SRV record was not returned")
+
+        # *** STEP 7 ***
+        # TH performs a query for the TXT record against the qname instance_qname.
+        # Verify TXT record is returned
+        self.step(7)
+        operational_record = await mdns.get_service_by_record_type(
+            service_name=instance_qname,
+            service_type=MdnsServiceType.OPERATIONAL.value,
+            record_type=DNSRecordType.TXT,
+            log_output=True,
+            load_from_cache=False
+        )
+
+        # Verify TXT record is returned and it contains values
+        txt_record_returned = operational_record.txt_record is not None and bool(operational_record.txt_record)
+        asserts.assert_true(txt_record_returned, "TXT record not returned or contains no values")
+
+        # *** STEP 8 ***
+        # TH performs a query for the AAAA record against the target listed in the SRV record.
+        # Verify AAAA record is returned
+        self.step(8)
+        
+        # domain = operational_record.server
```

Comment:
Is this the step that's giving you grief? Couldn't you just use a your MdnsAsyncServiceInfo class? This is just a query for AAAA. Name is the target from the SRV